src/bourso_api/src/types.rs

```diff
@@ -0,0 +1,254 @@
+            return Err(ValueError::TransferReason);
+        }
+        Ok(Self(t.into()))
+    }
+}
+impl FromStr for TransferReason {
+    type Err = ValueError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::new(s)
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, ValueEnum)]
+pub enum QuoteLength {
+    #[value(name = "1")]
+    D1,
+    #[value(name = "5")]
+    D5,
+    #[value(name = "30")]
+    D30,
+    #[value(name = "90")]
+    D90,
+    #[value(name = "180")]
+    D180,
+    #[value(name = "365")]
+    D365,
+    #[value(name = "1825")]
+    D1825,
+    #[value(name = "3650")]
+    D3650,
+}
+impl QuoteLength {
+    pub fn days(self) -> i64 {
+        match self {
+            QuoteLength::D1 => 1,
+            QuoteLength::D5 => 5,
+            QuoteLength::D30 => 30,
+            QuoteLength::D90 => 90,
+            QuoteLength::D180 => 180,
+            QuoteLength::D365 => 365,
+            QuoteLength::D1825 => 1825,
+            QuoteLength::D3650 => 3650,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize, ValueEnum)]
+pub enum OrderSide {
+    #[serde(rename = "B")]
+    Buy,
+    #[serde(rename = "S")]
+    Sell,
+}
+
+// TODO: support only 0 period for now, add support for other periods
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct QuotePeriod(i64);
+impl QuotePeriod {
+    pub fn new(v: i64) -> Result<Self, ValueError> {
+        if v == 0 {
+            Ok(Self(v))
+        } else {
+            Err(ValueError::QuotePeriod)
+        }
+    }
+    pub fn value(self) -> i64 {
+        self.0
+    }
+}
+impl FromStr for QuotePeriod {
+    type Err = ValueError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let v: i64 = s.trim().parse().map_err(|_| ValueError::QuotePeriod)?;
+        Self::new(v)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, AsRef, From, Into)]
+pub struct MfaCode(String);
+impl MfaCode {
+    pub fn new(s: &str) -> Result<Self, ValueError> {
+        let t = s.trim();
+        if (6..=12).contains(&t.len()) && t.chars().all(|c| c.is_ascii_digit()) {
+            Ok(Self(t.into()))
+        } else {
+            Err(ValueError::MfaCode)
+        }
+    }
+}
+impl FromStr for MfaCode {
+    type Err = ValueError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::new(s)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, AsRef, From, Into)]
+#[serde(try_from = "String", into = "String")]
+pub struct Password(String);
+impl Password {
+    pub fn new(s: &str) -> Result<Self, ValueError> {
+        let t = s.trim();
+        if !t.is_empty() {
+            Ok(Self(t.into()))
+        } else {
+            Err(ValueError::Password)
+        }
+    }
+}
+impl FromStr for Password {
+    type Err = ValueError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::new(s)
+    }
+}
```

Comment:
Debug on `Password` isn't a good idea IMO, would risk exposing the password to logs. For consistency, let's still implement Debug but with a redacted string.
```suggestion
#[derive(Clone, PartialEq, Eq, Hash, Serialize, Deserialize, AsRef, From, Into)]
#[serde(try_from = "String", into = "String")]
pub struct Password(String);
impl Password {
    pub fn new(s: &str) -> Result<Self, ValueError> {
        let t = s.trim();
        if !t.is_empty() {
            Ok(Self(t.into()))
        } else {
            Err(ValueError::Password)
        }
    }
}
impl FromStr for Password {
    type Err = ValueError;
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        Self::new(s)
    }
}
impl std::fmt::Debug for Password {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.write_str("Password(***REDACTED***)")
    }
}
```